test/unit/serializer/index.spec.ts

```diff
@@ -244,6 +305,78 @@ describe('index.ts', () => {
 			err = error;
 		}
 
+		expect(jsonDeserialize).toHaveBeenCalledTimes(1);
+		expect(gzipDeserialize).toHaveBeenCalledTimes(1);
 		expect(err).not.toBeUndefined();
 	});
+
+	it('should serialize enqueued when enqueue is used', async () => {
+		const target = new Serializer(new JsonStrategy(), { enqueue: 1 });
+		const call1 = jest.fn();
+		const call2 = jest.fn();
+		jest
+			.spyOn(target, 'serializeFactory' as any)
+			.mockImplementationOnce(
+				() => () => wait(10).then(call1).then(constant('result 1')),
+			)
+			.mockImplementationOnce(
+				() => () => wait(2).then(call2).then(constant('result 2')),
+			);
+
+		const result1 = target.serialize('data 1');
+		const result2 = await target.serialize('data 2');
+
+		expect(call1).toHaveBeenCalledBefore(call2);
+		expect(await result1).toBe('result 1');
+		expect(result2).toBe('result 2');
+	});
+
+	it('should serialize not enqueued when enqueue is used but the number of requests is lesser or equal than the pool size', async () => {
+		const target = new Serializer(new JsonStrategy(), { enqueue: 2 });
+		const call1 = jest.fn();
+		const call2 = jest.fn();
+		jest
+			.spyOn(target, 'serializeFactory' as any)
+			.mockImplementationOnce(
+				() => () => wait(10).then(call1).then(constant('result 1')),
+			)
+			.mockImplementationOnce(
+				() => () => wait(2).then(call2).then(constant('result 2')),
+			);
+
+		const result1 = target.serialize('data 1');
+		const result2 = await target.serialize('data 2');
+
+		expect(call2).toHaveBeenCalledBefore(call1);
```

Comment:
call2 is called before call1 because it takes less time to run and there is 2 queues, so, the two calls are distributed between them